docker-compose.dev.yml

```diff
@@ -2,7 +2,7 @@ version: '3'
 
 services:
     db:
-        image: postgres:11-alpine
+        image: postgres:12-alpine
```

Comment:
What was the issue in 11 prompting the switch to 12?

Reply:
Here are the tests that fail locally for me with postgres 11:

```
$ docker-compose exec web pytest posthog
========================================================================= test session starts =========================================================================
platform linux -- Python 3.8.8, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
django: settings: posthog.settings (from ini)
rootdir: /code, configfile: pytest.ini
plugins: mock-3.5.1, django-4.1.0, celery-4.4.2
collected 583 items

...

======================================================================== 583 passed in 58.39s =========================================================================

$ sed -i.bak 's/postgres:12-alpine/postgres:11-alpine/' docker-compose.dev.yml

$ docker-compose stop db ; docker-compose rm -fv db ; docker-compose up -d
Stopping posthog_db ... done
Going to remove posthog_db
Removing posthog_db ... done
posthog_redis is up-to-date
Creating posthog_db ... done
Recreating posthog_web_1    ... done
Recreating posthog_worker_1 ... done

$ docker-compose exec web pytest posthog
========================================================================= test session starts =========================================================================
platform linux -- Python 3.8.8, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
django: settings: posthog.settings (from ini)
rootdir: /code, configfile: pytest.ini
plugins: mock-3.5.1, django-4.1.0, celery-4.4.2
collected 583 items

...

============================================================================== FAILURES ===============================================================================
__________________________________________________________ TestAnnotation.test_query_annotations_by_datetime __________________________________________________________

self = <posthog.api.test.test_annotation.TestAnnotation testMethod=test_query_annotations_by_datetime>

    def test_query_annotations_by_datetime(self):

        Annotation.objects.create(
            team=self.team, created_by=self.user, content="hello_early", created_at="2020-01-04T13:00:01Z",
        )
        Annotation.objects.create(
            team=self.team, created_by=self.user, content="hello_later", created_at="2020-01-06T13:00:01Z",
        )
        response = self.client.get("/api/annotation/?before=2020-01-05").json()
        self.assertEqual(len(response["results"]), 2)
>       self.assertEqual(response["results"][1]["content"], "hello_early")
E       AssertionError: 'hello world!' != 'hello_early'
E       - hello world!
E       + hello_early

posthog/api/test/test_annotation.py:67: AssertionError
------------------------------------------------------------------------ Captured stderr call -------------------------------------------------------------------------
/usr/local/lib/python3.8/site-packages/django/db/models/fields/__init__.py:1365: RuntimeWarning: DateTimeField Annotation.created_at received a naive datetime (2020-01-05 00:00:00) while time zone support is active.
  warnings.warn("DateTimeField %s received a naive datetime (%s)"
____________________________________________________________________ TestEvent.test_filter_events _____________________________________________________________________

self = <posthog.api.test.test_event.TestEvent testMethod=test_filter_events>

    def test_filter_events(self):
        person_factory(
            properties={"email": "tim@posthog.com"},
            team=self.team,
            distinct_ids=["2", "some-random-uid"],
            is_identified=True,
        )

        event_factory(
            event="$autocapture",
            team=self.team,
            distinct_id="2",
            properties={"$ip": "8.8.8.8"},
            elements=[Element(tag_name="button", text="something"), Element(tag_name="div")],
        )
        event_factory(
            event="$pageview", team=self.team, distinct_id="some-random-uid", properties={"$ip": "8.8.8.8"}
        )
        event_factory(
            event="$pageview", team=self.team, distinct_id="some-other-one", properties={"$ip": "8.8.8.8"}
        )

        expected_queries = 4 if settings.PRIMARY_DB == RDBMS.CLICKHOUSE else 11

        with self.assertNumQueries(expected_queries):
            response = self.client.get("/api/event/?distinct_id=2").json()
>       self.assertEqual(
            response["results"][0]["person"],
            {"distinct_ids": ["2"], "is_identified": True, "properties": {"email": "tim@posthog.com"}},
        )
E       AssertionError: {'is_identified': True, 'distinct_ids': ['s[56 chars]om'}} != {'distinct_ids': ['2'], 'is_identified': Tr[42 chars]om'}}
E       - {'distinct_ids': ['some-random-uid'],
E       + {'distinct_ids': ['2'],
E          'is_identified': True,
E          'properties': {'email': 'tim@posthog.com'}}

posthog/api/test/test_event.py:45: AssertionError
_______________________________________________________________ TestDjangoPropertiesToQ.test_numerical ________________________________________________________________

self = <test_filter.TestDjangoPropertiesToQ testMethod=test_numerical>

    def test_numerical(self):
        event1 = event_factory(team=self.team, distinct_id="test", event="$pageview", properties={"$a_number": 5})
        event2 = event_factory(team=self.team, event="$pageview", distinct_id="test", properties={"$a_number": 6},)
        event_factory(
            team=self.team, event="$pageview", distinct_id="test", properties={"$a_number": "rubbish"},
        )
        filter = Filter(data={"properties": {"$a_number__gt": 5}})
        events = filter_events(filter, self.team)
        self.assertEqual(events[0]["id"], event2.pk)

        filter = Filter(data={"properties": {"$a_number": 5}})
        events = filter_events(filter, self.team)
        self.assertEqual(events[0]["id"], event1.pk)

        filter = Filter(data={"properties": {"$a_number__lt": 6}})
        events = filter_events(filter, self.team)
>       self.assertEqual(events[0]["id"], event1.pk)
E       AssertionError: 2725 != 2723

posthog/models/filters/test/test_filter.py:106: AssertionError
______________________________________________________________ TestDjangoTrends.test_breakdown_by_cohort ______________________________________________________________

self = <posthog.queries.test.test_trends.TestDjangoTrends testMethod=test_breakdown_by_cohort>

    def test_breakdown_by_cohort(self):
        person1, person2, person3, person4 = self._create_multiple_people()
        cohort = cohort_factory(name="cohort1", team=self.team, groups=[{"properties": {"name": "person1"}}])
        cohort2 = cohort_factory(name="cohort2", team=self.team, groups=[{"properties": {"name": "person2"}}])
        cohort3 = cohort_factory(
            name="cohort3",
            team=self.team,
            groups=[{"properties": {"name": "person1"}}, {"properties": {"name": "person2"}},],
        )
        action = action_factory(name="watched movie", team=self.team)
        action.calculate_events()

        with freeze_time("2020-01-04T13:01:01Z"):
            action_response = trends().run(
                Filter(
                    data={
                        "date_from": "-14d",
                        "breakdown": json.dumps([cohort.pk, cohort2.pk, cohort3.pk, "all"]),
                        "breakdown_type": "cohort",
                        "actions": [{"id": action.pk, "type": "actions", "order": 0}],
                    }
                ),
                self.team,
            )
            event_response = trends().run(
                Filter(
                    data={
                        "date_from": "-14d",
                        "breakdown": json.dumps([cohort.pk, cohort2.pk, cohort3.pk, "all"]),
                        "breakdown_type": "cohort",
                        "events": [{"id": "watched movie", "name": "watched movie", "type": "events", "order": 0,}],
                    }
                ),
                self.team,
            )
>       self.assertEqual(event_response[0]["label"], "watched movie - cohort1")
E       AssertionError: 'watched movie - cohort3' != 'watched movie - cohort1'
E       - watched movie - cohort3
E       ?                       ^
E       + watched movie - cohort1
E       ?                       ^

posthog/queries/test/test_trends.py:1764: AssertionError
======================================================================= short test summary info =======================================================================
FAILED posthog/api/test/test_annotation.py::TestAnnotation::test_query_annotations_by_datetime - AssertionError: 'hello world!' != 'hello_early'
FAILED posthog/api/test/test_event.py::TestEvent::test_filter_events - AssertionError: {'is_identified': True, 'distinct_ids': ['s[56 chars]om'}} != {'distinct_ids'...
FAILED posthog/models/filters/test/test_filter.py::TestDjangoPropertiesToQ::test_numerical - AssertionError: 2725 != 2723
FAILED posthog/queries/test/test_trends.py::TestDjangoTrends::test_breakdown_by_cohort - AssertionError: 'watched movie - cohort3' != 'watched movie - cohort1'
============================================================== 4 failed, 579 passed in 76.91s (0:01:16) ===============================================================
ERROR: 1
```

Reply:
Alright, should be good